cdc/sink/ddlsink/mysql/async_ddl.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/pingcap/log"
+	timodel "github.com/pingcap/tidb/pkg/parser/model"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/pkg/errors"
+	"go.uber.org/zap"
+)
+
+var checkRunningAddIndexSQL = `
+SELECT JOB_ID, JOB_TYPE, SCHEMA_STATE, SCHEMA_ID, TABLE_ID, STATE, QUERY
+FROM information_schema.ddl_jobs
+WHERE DB_NAME = "%s" 
+    AND TABLE_NAME = "%s"
+    AND JOB_TYPE LIKE "add index%%"
```

Comment:
what if it's an `drop index`, will it also takes a long time ?

Reply:
No, drop index is not a reorg ddl, [ref the doc](https://docs.pingcap.com/tidb/dev/ddl-introduction#:~:text=Currently%2C%20physical%20DDL%20statements%20only%20include%20ADD%20INDEX%20and%20lossy%20column%20type%20changes%20(such%20as%20changing%20from%20an%20INT%20type%20to%20a%20CHAR%20type).)